crates/router_env/src/root_span.rs

```diff
@@ -0,0 +1,70 @@
+//! Custom RootSpanBuilder tracing-actix-web
+
+use actix_web::{
+    body::MessageBody,
+    dev::{ServiceRequest, ServiceResponse},
+    http::StatusCode,
+    Error, HttpMessage, ResponseError,
+};
+use tracing::Span;
+use tracing_actix_web::{root_span, RootSpanBuilder};
+
+use crate::request_id::RequestId;
+
+/// Custom RootSpanBuilder that captures x-request-id header in spans
+#[derive(Debug)]
+pub struct CustomRootSpanBuilder;
+
+impl RootSpanBuilder for CustomRootSpanBuilder {
+    fn on_request_start(request: &ServiceRequest) -> Span {
+        // Extract the RequestId from extensions (set by RequestIdentifier middleware)
+        // We clone the string to avoid lifetime issues with the temporary Ref guard
+        let request_id = request
+            .extensions()
+            .get::<RequestId>()
+            .map(|id| id.to_string())
+            .unwrap_or_default();
+
+        root_span!(
+            level = crate::Level::INFO,
+            request,
+            request_id = request_id.as_str()
+        )
+    }
+
+    fn on_request_end<B: MessageBody>(span: Span, outcome: &Result<ServiceResponse<B>, Error>) {
+        match &outcome {
+            Ok(response) => {
+                if let Some(error) = response.response().error() {
+                    // use the status code already constructed for the outgoing HTTP response
+                    handle_error(span, response.status(), error.as_response_error());
+                } else {
+                    let code: i32 = response.response().status().as_u16().into();
+                    span.record("http.status_code", code);
+                    span.record("otel.status_code", "OK");
+                }
+            }
+            Err(error) => {
+                let response_error = error.as_response_error();
+                handle_error(span, response_error.status_code(), response_error);
+            }
+        };
+    }
```

Comment:
Are we required to explicitly set these fields? And/or do we necessarily need this implementation?

If there's no difference from the default implementation provided by the trait, we can remove this.